megamek/src/megamek/common/weapons/WeaponHandler.java

```diff
@@ -1555,8 +1556,11 @@ protected void handleEntityDamage(Entity entityTarget,
         }
         boolean isIndirect = wtype.hasModes()
                 && weapon.curMode().equals("Indirect");
+        IHex targetHex = game.getBoard().getHex(target.getPosition());
 
-        if (!isIndirect
+        //For indirect fire, remove leg hits only if target is in water partial cover
+        //Per TW errata for indirect fire
+        if ((!isIndirect || (isIndirect && targetHex.containsTerrain(Terrains.WATER)))
```

Comment:
This would work in most situations, but a mech standing in depth 0 water (or a superheavy in 0-1) receiving partial cover from terrain would incorrectly have the partial cover hit removed.

Reply:
Latest commit appears to fix this.